tools/coredumper/sof-coredump-reader.py

```diff
@@ -647,7 +673,7 @@ def __init__(self, args):
 		verbosePrint(self.core_dump.windowstart_process())
 
 		verbosePrint("# Location: " + str(self.file_info));
-		stack_base = self.core_dump.a[1] + 16
+		stack_base = self.core_dump.stack_ptr
```

Comment:
If we can a1 register is the stack ptr we have here, why we need to pass the stack top as the stack_ptr out?

Reply:
This existing +16 (stack area allocated by the arch_dump_regs_a) adjustment strongly depends on the .c implementation of the FW. 16 is just a difference between a1 (sp) set and dumped by the arch_dump_regs_a and the value of the stack pointer inside the dump_stack(). The magic number depends on how both functions are called and what is the relative a1 position in both functions, meaning there are multiple factors: area of stack being allocated in both call paths and how the functions are called. So it seems much more reliable to dump sp where the in-memory stack content is being dumped rather than guessing it from the sp inside arch_dump_regs_a and analysis of how the two relates to each other.

Reply:
One more advantage is that we are now not ignoring stack_rewind_frames variable